### ISSUE TITLE
`checkpatch.pl` warning about non-permalinks to Zulip

### ISSUE THREAD

**Issue by ojeda** [Rust-for-Linux/linux#1104]
Please see https://lore.kernel.org/rust-for-linux/CANiq72mrwBs_YTcBvge4ME5bwSLKbNaoFU+KZw3EfCTyGjiJ9w@mail.gmail.com/.

Please note that the `checkpatch.pl` maintainers will need to agree to the change.

Zulip apparently has support for permalinks incoming (or already present but not exposed through the UI). It would be ideal if the warning shows the correct link to use (if that is possible to infer from the other link) or at least point to the documentation (possibly https://chat.zulip.org/api/construct-narrow#narrows-that-use-ids) or how to do it in the UI (assuming it is ready in Zulip).

---

This requires submitting a proper patch to the LKML and the Rust for Linux mailing list. Please recall to test your changes (including generating the documentation if changed, running the Rust doctests if changed, etc.), to use a proper title for the commit, to sign your commit under the Developer's Certificate of Origin and to add a `Suggested-by:` tag and a `Link:` tag to this issue.

**Comment by BiscuitBobby** [comment]
Hello, I would like to give it a shot.

**Comment by ojeda** [comment]
Sure, please go ahead!

**Comment by BiscuitBobby** [comment]
Hey, wanted to clarify a few things:
- This includes all the subdomains of zulip.org and zulipchat.com and not just chat.zulip.org
- Stream links and message links are perma-links, ie., link should look like: 
  ```
  https://.../#narrow/stream/<anything>/ 
  ``` 
  or 
  ```
  https://.../#narrow/stream/<anything>/topic/<anything>/<with or near>/<number>
   ```

**Comment by ojeda** [comment]
> * This includes all the subdomains of zulip.org and zulipchat.com and not just chat.zulip.org

Hmm... If there are actual links that use those domains, I guess yes. However, I have only seen links like *.zulipchat.com, so unless we know they are used/generated by Zulip, I think it would be best to avoid them.

> * Stream links and message links are perma-links, ie., link should look like:

I am not sure what the new perma-links they are introducing are, but the ones we used so far have been always of the form:

https://rust-for-linux.zulipchat.com/#narrow/stream/x/topic/x/near/455637364

In order to shorten them, I usually replace the stream and topic name and leave an `x` there as above, since what counts is the number at the end.

If there is no number, and only a stream/topic name, then I think they are not perma-links, because if those change, the link is broken.

I would suggest testing the links to see if they are actual permalinks or not -- if needed, please feel free to test in our Zulip (or perhaps in a test instance they may have) creating a topic and then renaming it or similar.

Thanks!

**Comment by bjorn3** [comment]
chat.zulip.org is the zulip for development of zulip itself.

**Comment by BiscuitBobby** [comment]
Thanks for clearing it up for me, will test links from rust-for-linux.zulipchat.com

**Comment by BiscuitBobby** [comment]
I've sent in a patch to the mailing lists.

**Comment by ojeda** [comment]
https://lore.kernel.org/rust-for-linux/20240825221806.253575-1-simeddon@gmail.com/

**Comment by metaspace** [comment]
I don't understand how to use this feature. Is there a button somewhere, or do I have to manually write out something like this `https://rust-for-linux.zulipchat.com/#narrow/stream/x/topic/x/with/465040682` ?

For getting the message ID, am I supposed to cut it from the URL?

**Comment by bjorn3** [comment]
If you hover over a message and click on the three dots below each other, one of the menu options is "Copy link to message".

**Comment by ojeda** [comment]
That is what I do, yeah. I think Zulip is implementing something similar for topics too.

**Comment by metaspace** [comment]
> If you hover over a message and click on the three dots below each other, one of the menu options is "Copy link to message".

That would give me a link to a specific message. I think that if I want a permalink to a topic, I have to first get the link to the message and then rewrite the URL, right?

**Comment by ojeda** [comment]
No, it is a permalink; however, it would point to the first message, yes.

That is what I use -- I think it is good enough until they provide permalinks to topics.
